### PR TITLE
Add static host users to Terraform provider

### PR DESCRIPTION
**Pull request by atburke**
This change adds Terraform support for static host users.

Part of #42712.

Changelog: Added static host users to Terraform provider

### CODE REVIEW

File: docs/pages/reference/terraform-provider/resources/static_host_user.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+### Required
+
+- `metadata` (Attributes) metadata is resource metadata. (see [below for nested schema](#nested-schema-for-metadata))
+- `spec` (Attributes) spec is the static host user spec. (see [below for nested schema](#nested-schema-for-spec))
+
+### Optional
+
+- `sub_kind` (String) sub_kind is an optional resource sub kind, used in some resources.
+- `version` (String) version is the resource version. It must be specified. Supported values are: `v2`.
```
Comment (eriktate) [inline docs/pages/reference/terraform-provider/resources/static_host_user.mdx:51]: Should `version` this be in the `Required` section?

**Review by ptgott — APPROVED** [review]
Approving for docs

**Review by rosstimothy — COMMENTED** [review]
Can you please regenerate the code here to pick up https://github.com/gravitational/teleport/pull/46804?

File: docs/pages/reference/terraform-provider/data-sources/static_host_user.mdx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+### Optional
+
+- `sub_kind` (String) sub_kind is an optional resource sub kind, used in some resources.
+
+### Nested Schema for `metadata`
+
+Required:
+
+- `name` (String) name is an object name.
```
[the remark this answers is not in the document]
Comment (atburke) [reply docs/pages/reference/terraform-provider/data-sources/static_host_user.mdx:30]: Not without changes to `protoc-gen-terraform`. That text comes straight from the headerv1 metadata proto.

File: docs/pages/reference/terraform-provider/data-sources/static_host_user.mdx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Optional:
+
+- `description` (String) description is object description.
+- `expires` (String)
```
Comment (tigrato) [inline docs/pages/reference/terraform-provider/data-sources/static_host_user.mdx:35]: not related but we should update expires description with the correct format

**Review by tigrato — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**